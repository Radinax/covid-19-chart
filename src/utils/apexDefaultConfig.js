export default function(title, isHorizontal) {
  console.log('ishir', isHorizontal)
  return {
    options: {
      chart: {
        //background: '#f4f4f4',
        // foreColor: '#333'
      },
      xaxis: {
        categories: []
      },
      plotOptions: {
        bar: {
          horizontal: isHorizontal, // if true it would be horizontal
        }
      },
      fill: {
        // colors: ['#f44336']
      },
      dateLabels: {
        enabled: true,
        position: 'top',
        style: {
          fontSize: '40px',
          fontWeight: 'bold',
        }
      },
      title: {
        text: title,
        align: 'center',
        margin: 20,
        offsetY: 20,
        style: {
          fontSize: '25px'
        }
      }
    },
    series: [{
      name: 'Cases',
      data: []
    }]
  }
}
