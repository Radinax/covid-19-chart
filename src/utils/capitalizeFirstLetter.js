export const capitalizeFirstLetter = (string) => {
  return string[0].toUpperCase() + string.slice(1);
}
