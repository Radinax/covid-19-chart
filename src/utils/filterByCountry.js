export const filterByCountry = (arr, country) => arr.filter(o => o.country === country)[0]
