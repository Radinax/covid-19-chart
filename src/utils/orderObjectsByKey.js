export const orderObjectByKey = (unOrderedObject, orderedKeys) => {
  const orderedData = orderedKeys.reduce((o, index) => {
    o[index] = unOrderedObject[index]
    return o
  }, {})
  return orderedData
}
