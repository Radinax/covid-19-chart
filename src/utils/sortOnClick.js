export const sortOnClick = (isAsc, sortKey, data) => {
  const sortedData = data.sort((a,b) => {
    if (isAsc) {
      if (a[sortKey] > b[sortKey]) return -1
      if (a[sortKey] < b[sortKey]) return 1
      return 0;
    } else {
      if (a[sortKey] > b[sortKey]) return 1
      if (a[sortKey] < b[sortKey]) return -1
      return 0;
    }
  })
  return sortedData
}
