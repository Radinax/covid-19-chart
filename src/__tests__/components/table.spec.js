import React from 'react'
import { render } from '@testing-library/react'
import '@testing-library/jest-dom/extend-expect'
import '@testing-library/jest-dom'
import Table from '../../components/table'
import { capitalizeFirstLetter } from '../../utils/capitalizeFirstLetter'
import { sortOnClick } from '../../utils/sortOnClick'

const headerData = ['country', 'recovered', 'deaths']
const bodyData = [
  { country: 'A', recovered: '1', deaths: '2' },
  { country: 'B', recovered: '2', deaths: '1' },
]

describe('if Select is working correctly', () => {
  it('renders', () => {
    const { asFragment } = render(<Table headerData={headerData} bodyData={bodyData} />)
    expect(asFragment()).toMatchSnapshot()
  })

  it('has buttons elements in the header', () => {
    const { container } = render(<Table headerData={headerData} bodyData={bodyData} />)
    expect(container.querySelector('button')).toBeTruthy()
  })

  test('headerData props is mapping correctly into thead', () => {
    const { container } = render(<Table headerData={headerData} bodyData={bodyData} />)
    const expectedText = headerData.reduce((prev, next) => {
      return capitalizeFirstLetter(prev) + capitalizeFirstLetter(next)
    })
    expect(container.querySelector('thead')).toHaveTextContent(expectedText)
  })

  test('bodyData is showing correct values in its row', () => {
    const { container } = render(<Table headerData={headerData} bodyData={bodyData} />)
    const tableBody = container.querySelector('tbody')
    const tableRow = tableBody.children
    // First cell of our first row inside our table body
    expect(tableRow[0].children[0]).toHaveTextContent(bodyData[0].country)
    // Second cell of our first row inside our table body
    expect(tableRow[0].children[1]).toHaveTextContent(bodyData[0].recovered)
    // Third cell of our first row inside our table body
    expect(tableRow[0].children[2]).toHaveTextContent(bodyData[0].deaths)
    // First cell of our second row inside our table body
    expect(tableRow[1].children[0]).toHaveTextContent(bodyData[1].country)
    // Second cell of our second row inside our table body
    expect(tableRow[1].children[1]).toHaveTextContent(bodyData[1].recovered)
    // Third cell of our second row inside our table body
    expect(tableRow[1].children[2]).toHaveTextContent(bodyData[1].deaths)
  })

  test('asc data', () => {
    const sortedData = sortOnClick(true, headerData[1], [...bodyData])
    const { asFragment, container } = render(<Table headerData={headerData} bodyData={sortedData} />)
    const tableBody = container.querySelector('tbody')
    const tableRow = tableBody.children
    // It should be the opposite of the previous unit test since we show ASC values
    expect(tableRow[0].children[0]).toHaveTextContent(bodyData[1].country)
    expect(tableRow[0].children[1]).toHaveTextContent(bodyData[1].recovered)
    expect(tableRow[0].children[2]).toHaveTextContent(bodyData[1].deaths)
    expect(tableRow[1].children[0]).toHaveTextContent(bodyData[0].country)
    expect(tableRow[1].children[1]).toHaveTextContent(bodyData[0].recovered)
    expect(tableRow[1].children[2]).toHaveTextContent(bodyData[0].deaths)
    expect(asFragment()).toMatchSnapshot()
  })

  test('desc data', () => {
    const sortedData = sortOnClick(false, headerData[1], [...bodyData])
    const { asFragment, container } = render(<Table headerData={headerData} bodyData={sortedData} />)
    const tableBody = container.querySelector('tbody')
    const tableRow = tableBody.children
    // It should be the opposite of the previous unit test since we show ASC values
    expect(tableRow[0].children[0]).toHaveTextContent(bodyData[0].country)
    expect(tableRow[0].children[1]).toHaveTextContent(bodyData[0].recovered)
    expect(tableRow[0].children[2]).toHaveTextContent(bodyData[0].deaths)
    expect(tableRow[1].children[0]).toHaveTextContent(bodyData[1].country)
    expect(tableRow[1].children[1]).toHaveTextContent(bodyData[1].recovered)
    expect(tableRow[1].children[2]).toHaveTextContent(bodyData[1].deaths)
    expect(asFragment()).toMatchSnapshot()
  })
})
