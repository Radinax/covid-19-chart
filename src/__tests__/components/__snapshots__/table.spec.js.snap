// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`if Select is working correctly asc data 1`] = `
<DocumentFragment>
  <table
    data-testid="table"
    style="margin-top: 2rem; padding-right: 1rem;"
  >
    <thead>
      <tr>
        <th
          style="text-align: start;"
        >
          <button>
            Country
          </button>
        </th>
        <th
          style="text-align: start;"
        >
          <button>
            Recovered
          </button>
        </th>
        <th
          style="text-align: start;"
        >
          <button>
            Deaths
          </button>
        </th>
      </tr>
    </thead>
    <tbody>
      <tr>
        <td
          style="text-align: start;"
        >
          B
        </td>
        <td
          style="text-align: end;"
        >
          2
        </td>
        <td
          style="text-align: end;"
        >
          1
        </td>
      </tr>
      <tr>
        <td
          style="text-align: start;"
        >
          A
        </td>
        <td
          style="text-align: end;"
        >
          1
        </td>
        <td
          style="text-align: end;"
        >
          2
        </td>
      </tr>
    </tbody>
  </table>
</DocumentFragment>
`;

exports[`if Select is working correctly desc data 1`] = `
<DocumentFragment>
  <table
    data-testid="table"
    style="margin-top: 2rem; padding-right: 1rem;"
  >
    <thead>
      <tr>
        <th
          style="text-align: start;"
        >
          <button>
            Country
          </button>
        </th>
        <th
          style="text-align: start;"
        >
          <button>
            Recovered
          </button>
        </th>
        <th
          style="text-align: start;"
        >
          <button>
            Deaths
          </button>
        </th>
      </tr>
    </thead>
    <tbody>
      <tr>
        <td
          style="text-align: start;"
        >
          A
        </td>
        <td
          style="text-align: end;"
        >
          1
        </td>
        <td
          style="text-align: end;"
        >
          2
        </td>
      </tr>
      <tr>
        <td
          style="text-align: start;"
        >
          B
        </td>
        <td
          style="text-align: end;"
        >
          2
        </td>
        <td
          style="text-align: end;"
        >
          1
        </td>
      </tr>
    </tbody>
  </table>
</DocumentFragment>
`;

exports[`if Select is working correctly renders 1`] = `
<DocumentFragment>
  <table
    data-testid="table"
    style="margin-top: 2rem; padding-right: 1rem;"
  >
    <thead>
      <tr>
        <th
          style="text-align: start;"
        >
          <button>
            Country
          </button>
        </th>
        <th
          style="text-align: start;"
        >
          <button>
            Recovered
          </button>
        </th>
        <th
          style="text-align: start;"
        >
          <button>
            Deaths
          </button>
        </th>
      </tr>
    </thead>
    <tbody>
      <tr>
        <td
          style="text-align: start;"
        >
          A
        </td>
        <td
          style="text-align: end;"
        >
          1
        </td>
        <td
          style="text-align: end;"
        >
          2
        </td>
      </tr>
      <tr>
        <td
          style="text-align: start;"
        >
          B
        </td>
        <td
          style="text-align: end;"
        >
          2
        </td>
        <td
          style="text-align: end;"
        >
          1
        </td>
      </tr>
    </tbody>
  </table>
</DocumentFragment>
`;
